Userland/Libraries/LibCore/System.cpp

```diff
@@ -36,6 +36,12 @@ ErrorOr<void> pledge(StringView promises, StringView execpromises)
     HANDLE_SYSCALL_RETURN_VALUE("pledge"sv, rc, {});
 }
 
+ErrorOr<void> revoke(StringView promises)
+{
+    (void)promises;
```

Comment:
Another disadvantage of having `revoke()` instead of just calling pledge with more and more restricted surface area, is that now we need a separate syscall to implement revoke. That means we have yet another syscall that needs to be audited / fuzzed and that has to be able to be called no matter what is pledged, as you always need to be able to revoke theoretically..

Reply:
See https://github.com/SerenityOS/serenity/issues/11140#issuecomment-991602533 : everything can internally fall back on `pledge()`.

Reply:
It doesn't seem wise to have libc tracking or aggregating pledges. It's just more moving parts that can fail?

Reply:
Sorry, my communication was ambiguous: `Core::System::pledge()` and `Core::System::revoke()` track the promises and both eventually delegate a `const char*` to the system call `pledge()`.